src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift

```diff
@@ -44,4 +46,107 @@ final class TicketValidation: TicketValidating {
 
 	}
 
+	// swiftlint:disable:next cyclomatic_complexity
+	func requestIdentityDocumentOfTheValidationDecorator(
+		urlString: String,
+		completion:
+		@escaping (Result<DecoratorIdentityDocument, DecoratorServiceIdentityDocumentError>) -> Void
+	) {
+		guard let url = URL(string: urlString) else {
+			Log.error("URL cant be constructed from input string", log: .ticketValidationDecorator)
+			return
+		}
+		let resource = DecoratorIdentityDocumentResource(url: url)
+		restServiceProvider.load(resource) { result in
+			switch result {
```

Comment:
For the hole switch I'd like to see some unit tests. But how test this big thing? Extract all tech-spec-points into separate functions, which can be tested then independently.

Reply:
I'm wondering if this whole function could be extracted into a separate struct or class. In the end it will be called from the `initialize()` func, so it would make sense to extract and unit test it somewhere else and just use it in `TicketValidation`